app/controllers/sessions_controller.rb

```diff
@@ -2,20 +2,38 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-    user = User.find_by email: params.dig(:session, :email)&.downcase
-    if user.try(:authenticate, params.dig(:session, :password))
-      reset_session
-      log_in user
-      params.dig(:session, :remember_me) == "1" ? remember(user) : forget(user)
-      redirect_to user, status: :see_other
+    user = find_user_by_email
+    if user_authenticated? user
+      handle_authenticated_user user
     else
-      flash.now[:danger] = t ".message.invalid_email_password_combination"
-      render :new, status: :unprocessable_entity
+      handle_failed_authentication
     end
   end
 
   def destroy
     log_out
     redirect_to root_url, status: :see_other
   end
+
+  private
+
+  def find_user_by_email
+    User.find_by email: params.dig(:session, :email)&.downcase
```

Comment:
thêm code handle user bị nil luôn cu
```suggestion
    @user = User.find_by email: params.dig(:session, :email)&.downcase
    
    Handle user nil
```

rồi dùng before_action để call nha 
ko cần gọi trong method đâu